### ISSUE TITLE
generate-term for integers only generates small naturals

### ISSUE THREAD

**Issue by dvanhorn** [racket/redex#191]
I've noticed that generate-term generates small natural numbers with very high probability (and I believe it never generates negative integers, even if given a large size value). 

Here's a transcript:
```racket
#lang racket
(require redex)
(define-language L)
(define (rand-test g i)
  (set-count (for/set ([_ i]) (g 100000)))) ; size = 100K
```
```
> (rand-test (generate-term L integer) 100)
8
> (rand-test (generate-term L integer) 1000)
11
> (rand-test (generate-term L integer) 10000)
13
> (rand-test (generate-term L integer) 100000)
18
```
Things are better for variables and strings:
```
> (rand-test (generate-term L variable) 100000)
6773
> (rand-test (generate-term L string) 100000)
53733
```
I think it's good that small numbers are favored, but it should be possible to get some larger results.

**Comment by rfindler** [comment]
I think you want to use the `#:attempts` expr to get it to try "further out" things. For example:

```
> (rand-test
   (λ (s) (generate-term L integer s #:attempt-num 100000))
   100)
31
```

And I'm pretty sure you know this already but it may make sense for you to consider `contract-random-generate` instead of redex, depending on what you're doing.

FWIW, `redex-check` uses that so it finds a counterexample pretty quickly to this one:

```
#lang racket
(require redex)
(define-language L)
(redex-check
 L
 integer
 (not (negative? (term integer))))
```

Of course, that said, tweaks to the way things work are welcome, but tuning a generic random generator can be a pretty frustrating activity.... :(

**Comment by dvanhorn** [comment]
Thanks -- I wonder if you would consider exposing what `redex-check` uses?  After trying `generate-term` for a while, I moved to using `contract-random-generate` by hand translating language grammars to flat contracts.  It sounds like `redex-check` is doing something similar automatically.

I also noticed that `contract-random-generate` doesn't work for any mutually recursive flat contracts, even one that is equivalent to a non-mutually rec. one, for example:

```racket
(contract-random-generate
 (flat-rec-contract t #t)) ; always #t

(contract-random-generate
 (flat-murec-contract ([t #t]) t)) ; unable to construct
```

Is this expected?
